napari/layers/base/base.py

```diff
@@ -1522,14 +1556,27 @@ def _world_to_layer_dims(
         ndarray
             The corresponding layer dimensions with the same ordering as the given world dimensions.
         """
-        offset = ndim_world - self.ndim
-        order = np.array(world_dims)
-        if offset == 0:
-            return order
-        if offset < 0:
-            return np.concatenate(np.arange(-offset), order - offset)
-
-        return order[order >= offset] - offset
+        return self._world_to_layer_dims_impl(
+            world_dims, ndim_world, self.ndim
+        )
+
+    @staticmethod
+    def _world_to_layer_dims_impl(
+        world_dims: npt.NDArray, ndim_world: int, ndim: int
+    ):
```

Comment:
if you're going to make it a static method I'd rather it be placed with other utility functions

Reply:
Making it static is only useful for testing it allows me to test it without having to create a Layer. It's not really useful in utils as it will work only with Layer, and it's not really a util function. 

The point of static method is also namespacing. I don't quite agree that because a method is static it will be better standalone. If I put it in utils it will need to be made 1/2 public which I don't like.

And right now it is an implementation detail of `_world_to_layer_dims` which itself is private. 

If you still really want me to I can, but I don't think it's a good choice.

Reply:
More particularly, many of the methods of Layer are `@abstractmethod`, so I can't test the method directly without having to instantiate a subclass, which is counter to unit test to test only what you wish to test.